### PR TITLE
Various doc updates

### PR DESCRIPTION
**Pull request by michaeljohnalbers**
Several updates are made under this PR.

* I alphabetized a lot of the section of the docs menus. Not all as some made more sense to leave as-is (for instance the Query section).
* Documentation was added for the following statements: CALL, SET, VACUUM, USE
* Documentation was added for specifying S3 configuration values in the S3 URL
* A few other small, miscellaneous changes.

I didn't update the search files as I'm going to be lazy and let the Github action handle that.

### CODE REVIEW

File: docs/sql/statements/call.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+selected: Documentation/SQL/Call
+expanded: SQL
+railroad: statements/call.js
+---
+
+The `CALL` statement invokes the given table function and returns the results.
+
+### Examples
+```sql
+-- Inoke the 'duckdb_functions' table function.
```
Comment (Mytherin) [inline docs/sql/statements/call.md:13]: typo: `Invoke`

File: docs/sql/statements/vacuum.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
+---
+layout: docu
+title: Vacuum
+selected: Documentation/SQL/Vacuum
+expanded: SQL
+railroad: statements/vacuum.js
+---
+
+The `VACUUM` statement garbage collects DuckDB.
```
Comment (Mytherin) [inline docs/sql/statements/vacuum.md:9]: The `VACUUM` statement currently does not garbage collect anything. It was initially added as a nop operation for Postgres compatibility - but now `VACUUM ANALYZE` can be used to recompute statistics if they have gone out of date due to deletions or updates made to the table.

We do plan to add a working `VACUUM` in the future, however.

File: js/statements/use.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+function GenerateUse(options = {}) {
+    return Diagram([
+        AutomaticStack([
+            Keyword("USE"),
+            Expression("database-name"),
+            Optional(Expression(".schema-name"), "skip")
```
Comment (Mytherin) [inline js/statements/use.js:6]: The `.` should be a separate `Keyword`

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! Looks good to me. Some comments below:

File: docs/sql/statements/call.md (hunk 1 above)
Comment (michaeljohnalbers) [reply docs/sql/statements/call.md]: Fixed on https://github.com/duckdb/duckdb-web/pull/688/commits/355fc58fed2093e4fb6dfc57b2e3c80f16cc126b

File: js/statements/use.js (hunk 3 above)
Comment (michaeljohnalbers) [reply js/statements/use.js]: Fixed on https://github.com/duckdb/duckdb-web/pull/688/commits/355fc58fed2093e4fb6dfc57b2e3c80f16cc126b

**Comment by Mytherin** [comment]
Thanks! LGTM

**Pull request merged**